File: Firefly/UI/ImageResult/ImageResult.js
function initUI() {
    sketchup.get_result_files();

    document.getElementById('file_select').addEventListener('change', e => updatePreview());
}

function showImage() {
    let options = {
        'file_name': getElementValue('file_select'),
        'type': getElementValue('type_select'),
        'falsecolor_scale': getElementValue('falsecolor_input'),
        'falsecolor_palette': document.querySelector('[name=falsecolor_radio]:checked').value
    };

    sketchup.create_image(options);
}

function updateFileSelect(resultFiles) {
    let select = document.getElementById('file_select');
    
    for (_ in select.options) {
        select.remove(0);
    }

    if (resultFiles.length == 0) {
        let option = document.createElement('option');
        option.text = 'No result files found';
        select.add(option);
    }

    for (const rf of resultFiles) {
        let option = document.createElement('option');
        option.text = rf;
        select.add(option);
    }

    updatePreview();
}

function typeChanged() {
    let type = getElementValue('type_select');
    document.getElementById('falsecolor_div').hidden = (type != 'False Color');
    console.log(type);
}

function getElementValue(id) {
    return document.getElementById(id).value
}

function updatePreview(file) {
    if (file == null){
        let file = getElementValue('file_select');
        sketchup.get_preview(file);
    }
    else {
        preview_img = document.getElementById('preview_img');
        preview_img.src = file;
    }
}

File: Firefly/UI/ImageResult/ImageResult.rb
module Firefly
  module Dialog
    module ImageResult
      class << self
        def show_dialog
          dialog = create_dialog

          dialog.add_action_callback('create_image') do |_context, options|
            create_image options
          end

          dialog.add_action_callback('get_result_files') do |_context|
            result_files = fetch_result_files
            dialog.execute_script "updateFileSelect(#{result_files})"
          end

          dialog.add_action_callback('get_preview') do |_context, file|
            preview_file = "#{File.basename(file, '.hdr')}.bmp"
            preview_file = File.join(Directory.preview_dir, preview_file)
            dialog.execute_script "updatePreview('#{preview_file}')"
          end

          dialog.show
        end

        private

        def create_dialog
          hmtl_file = File.join(__dir__, 'ImageResult.html')

          dialog = UI::HtmlDialog.new(
            dialog_title: 'Image Results',
            width: 300,
            height: 600,
            resizable: false,
            style: UI::HtmlDialog::STYLE_DIALOG
          )
          dialog.set_file hmtl_file
          dialog
        end

        def fetch_result_files
          results_dir = Directory.results_dir
          files = Dir["#{results_dir}/*"]
          files.map { |f| File.basename f }
        end

        def create_image(options)
          case options['type']
          when 'Human Conditioned'
            create_pcond_image options
          when 'False Color'
            create_false_color_image options
          when 'Raw Image'
            create_raw_image options
          end
        end

        def create_pcond_image(options)
          ResultHandler.pcond_image options['file_name']
        end

        def create_false_color_image(options)
          ResultHandler.false_color_image(*options.values_at('file_name', 'falsecolor_scale', 'falsecolor_palette'))
        end

        def create_raw_image(options)
          ResultHandler.raw_image options['file_name']
        end
      end
    end
  end
end
